src/storage/s3.rs

```diff
@@ -190,6 +191,47 @@ pub(crate) mod tests {
         });
     }
 
+    #[test]
+    fn test_get_too_big() {
```

Comment:
Buffer testing here as well?

Reply:
What do you mean?

Reply:
Well, the sized buffer is used indirectly by the things being tested, and so making sure that the sized buffer preforms as expected in a real scenario seems logical to me